instrumentation/excon/test/opentelemetry/instrumentation/excon/instrumentation_test.rb

```diff
@@ -150,4 +151,143 @@
+      _(span.attributes['net.peer.port']).must_equal(80)
+    end
+  end
+
+  # NOTE: WebMock introduces an extra HTTP request span due to the way the mocking is implemented.
+  describe '#connect' do
+    before do
+      instrumentation.install
+    end
+
+    it 'emits span on connect' do
+      WebMock.allow_net_connect!
+
+      TCPServer.open('localhost', 0) do |server|
+        Thread.start do
+          server.accept
+        rescue IOError
+          nil
+        end
+
+        port = server.addr[1]
+
+        _(-> { Excon.get("http://localhost:#{port}/example", read_timeout: 0) }).must_raise(Excon::Error::Timeout)
+      end
+
+      _(exporter.finished_spans.size).must_equal(3)
+      _(span.name).must_equal 'connect'
+      _(span.attributes['net.peer.name']).must_equal('localhost')
+      _(span.attributes['net.peer.port']).wont_be_nil
+    ensure
+      WebMock.disable_net_connect!
+    end
+
+    it 'captures errors' do
+      WebMock.allow_net_connect!
+
+      _(-> { Excon.get('http://invalid.com:99999/example') }).must_raise
+
+      _(exporter.finished_spans.size).must_equal(3)
+      _(span.name).must_equal 'connect'
+      _(span.attributes['net.peer.name']).must_equal('invalid.com')
+      _(span.attributes['net.peer.port']).must_equal(99_999)
+
+      span_event = span.events.first
+
+      _(span_event.name).must_equal 'exception'
+      _(span_event.attributes['exception.type']).must_equal(SocketError.name)
+    ensure
+      WebMock.disable_net_connect!
+    end
+
+    it '[BUG] fails to emit an HTTP CONNECT span when connecting through an SSL proxy for an HTTP service' do
+      WebMock.allow_net_connect!
+
+      _(-> { Excon.get('http://localhost/', proxy: 'https://proxy_user:proxy_pass@localhost') }).must_raise(Excon::Error::Socket)
+
+      _(exporter.finished_spans.size).must_equal(3)
+      _(span.name).must_equal 'connect'
+      _(span.kind).must_equal(:internal)
+      _(span.attributes['net.peer.name']).must_equal('localhost')
+      _(span.attributes['net.peer.port']).must_equal(443)
+    ensure
+      WebMock.disable_net_connect!
+    end
+
+    it 'emits an HTTP CONNECT span when connecting through an SSL proxy' do
+      WebMock.allow_net_connect!
+
+      _(-> { Excon.get('https://localhost/', proxy: 'https://proxy_user:proxy_pass@localhost') }).must_raise(Excon::Error::Socket)
+
+      _(exporter.finished_spans.size).must_equal(3)
+      _(span.name).must_equal 'HTTP CONNECT'
+      _(span.kind).must_equal(:client)
+      _(span.attributes['net.peer.name']).must_equal('localhost')
+      _(span.attributes['net.peer.port']).must_equal(443)
+    ensure
+      WebMock.disable_net_connect!
+    end
+
+    it 'emits a "connect" span when connecting through an non-ssl proxy' do
+      WebMock.allow_net_connect!
+
+      _(-> { Excon.get('http://localhost', proxy: 'https://proxy_user:proxy_pass@localhost') }).must_raise(Excon::Error::Socket)
+
+      _(exporter.finished_spans.size).must_equal(3)
```

Comment:
is there anything interesting about the other 2 spans that we would want to review/check?

Reply:
The last span is because of the way WebMock mocks requests in Excon, so we can always ignore that one. The middle span, we could check that it recorded the exception but we already did that outside of the connect context.

Reply:
I added assertions on this anyways just to be thorough.